core/capabilities/ccip/ccipevm/helpers.go

```diff
@@ -74,28 +37,19 @@ func ABITypeOrPanic(t string) abi.Type {
 	return abiType
 }
 
-// Decodes the given bytes into a uint32, based on the encoding of destGasAmount in FeeQuoter.sol
-func decodeTokenDestGasOverhead(destExecData []byte) (uint32, error) {
-	ifaces, err := TokenDestGasOverheadABI.UnpackValues(destExecData)
-	if err != nil {
-		return 0, fmt.Errorf("abi decode TokenDestGasOverheadABI: %w", err)
-	}
-	_, ok := ifaces[0].(uint32)
-	if !ok {
-		return 0, fmt.Errorf("expected uint32, got %T", ifaces[0])
-	}
-	return ifaces[0].(uint32), nil
-}
-
 // CCIPMsgToAny2EVMMessage converts a ccipocr3.Message object to an offramp.InternalAny2EVMRampMessage object.
 // These are typically used to create the execution report for EVM.
-func CCIPMsgToAny2EVMMessage(msg ccipocr3.Message) (offramp.InternalAny2EVMRampMessage, error) {
+func CCIPMsgToAny2EVMMessage(msg ccipocr3.Message, codec ccipcommon.ExtraDataCodec) (offramp.InternalAny2EVMRampMessage, error) {
 	var tokenAmounts []offramp.InternalAny2EVMTokenTransfer
 	for _, rta := range msg.TokenAmounts {
-		destGasAmount, err := abiDecodeUint32(rta.DestExecData)
+		decodedMap, err := codec.DecodeTokenAmountDestExecData(rta.DestExecData, msg.Header.SourceChainSelector)
 		if err != nil {
 			return offramp.InternalAny2EVMRampMessage{}, fmt.Errorf("failed to decode dest gas amount: %w", err)
 		}
+		destGasAmount, err := extractDestGasAmountFromMap(decodedMap)
+		if err != nil {
+			return offramp.InternalAny2EVMRampMessage{}, fmt.Errorf("failed to extract dest gas amount from decodec map: %w", err)
+		}
 
 		tokenAmounts = append(tokenAmounts, offramp.InternalAny2EVMTokenTransfer{
 			SourcePoolAddress: common.LeftPadBytes(rta.SourcePoolAddress, 32),
```

Comment:
Isn't this also wrong if Solana was the source?

Reply:
The LeftPadBytes? It shouldn't have any effect on Solana address since it's already 32 bytes